test/jwt.test.js

```diff
@@ -0,0 +1,341 @@
+    var EXPECTED_ERROR_MESSAGE = 'Audience (aud) payload mismatch in the ID token; expected "expectedAudience" but was not one of "tokens-test-123, external-test-999"';
+    try {
+      expectedOptions.aud = 'expectedAudience';
+      jwt.verify(generateJWT({}), expectedOptions)
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when azm payload not found when aud has multiple values', function (done) {
+    var EXPECTED_ERROR_MESSAGE = 'Authorized Party (azp) payload must be a string present in the ID token when Audience (aud) payload has multiple values';
+    try {
+      var token = generateJWT({ azp: undefined });
+      jwt.verify(token, expectedOptions)
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when azm payload doesnt match the expected aud', function (done) {
+    var EXPECTED_ERROR_MESSAGE = 'Authorized Party (azp) payload mismatch in the ID token; expected "external-test-999", found "tokens-test-123"';
+    try {
+      var token = generateJWT({});
+      expectedOptions.aud = expectations.clientIdAlt;
+      jwt.verify(token, expectedOptions)
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when nonce is in options, but missing from payload ', function (done) {
+    var EXPECTED_ERROR_MESSAGE = 'Nonce (nonce) payload must be a string present in the ID token';
+    try {
+      var token = generateJWT({ nonce: undefined });
+
+      jwt.verify(token, expectedOptions)
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when nonce payload doesnt match nonce expected', function (done) {
+    var EXPECTED_ERROR_MESSAGE = 'Nonce (nonce) payload mismatch in the ID token; expected "noncey", found "notExpectedNonce"';
+    try {
+      var token = generateJWT({ nonce: 'notExpectedNonce' });
+      expectedOptions.nonce = 'noncey';
+      jwt.verify(token, expectedOptions)
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when auth_time is not a number', function (done) {
+    var EXPECTED_ERROR_MESSAGE = 'Authentication Time (auth_time) payload must be a number present in the ID token when Max Age (max_age) is specified';
+    try {
+      var token = generateJWT({ auth_time: 'not a number' });
```

Comment:
```suggestion
      var token = generateJWT({ auth_time: undefined });
```

More likely to be missing than the wrong type